controllers/azurecluster_controller.go

```diff
@@ -230,6 +236,74 @@ func (r *AzureClusterReconciler) reconcileNormal(ctx context.Context, clusterSco
 		params.BastionIP = clusterAnnotations[BastionHostIPAnnotation]
 	}
 
+	// `azureCluster.spec.networkSpec.apiServerLB.privateLinks` is the current identifier
+	// for private DNS zone creation in the management cluster
```

Comment:
Instead of doing the decision within the nomral reconciliation, an alternative approach could be an additional reconciliation loop where we only reconcile on clusters where the label `giantswarm.io/cluster-type: private` match (label doesn't exist yet).